### PR TITLE
[RW-7106][risk=no] Pyspark code snippets

### PR DESCRIPTION
**Pull request by calbach**
I may not merge this, just for initial review.

### CODE REVIEW

File: api/docs/pyspark-usage.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+unzips it, then counts the lines.
+
+```python
+import os
+import subprocess
+
+input_files = !gsutil ls gs://fc-aou-test-datasets-registered/6/wgs/vcf/merged
+input_files = input_files[:5]
+
+def wc(path):
```
[the remark this answers is not in the document]
Comment (ericsong) [reply api/docs/pyspark-usage.md:71]: nit: rename the argument to `input_file` or `input_file_path` so it's more clear that its coming from the `input_files` argument?

File: api/docs/pyspark-usage.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+sc.parallelize(input_files).map(wc).collect()
+```
+
+## Run an arbitrary bash script over a collection of files
+
+```bash
+%%bash
+
+cat > my_script.sh << EOF
```
Comment (ericsong) [inline api/docs/pyspark-usage.md:88]: nit: I didn't know this syntax for piping in text so I was a little confused at first as to what it was doing. I'm wondering if we should just go with the more verbose method of building the file line by line with `echo`. Alternatively, a good comment should work.

File: api/docs/pyspark-usage.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+#!/bin/bash
+
+echo "processing file" \$1
+EOF
+
+chmod +x my_script.sh
```
Comment (ericsong) [inline api/docs/pyspark-usage.md:94]: nit: comment explaining what this does

File: api/docs/pyspark-usage.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+./my_script.sh foo.txt
```
Comment (ericsong) [inline api/docs/pyspark-usage.md:95]: nit: comment explaining what this does and that its not necessary (I think? it's just testing it right)

File: api/docs/pyspark-usage.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+## Write to workspace bucket from workers
+
+```python
+# This is necessary to make hadoop fs work, even though GCS doesn't have the concept of directories.
+! hadoop fs -mkdir -p “${WORKSPACE_BUCKET}/spark-test/outputs”
+
+import os
+import subprocess
+
+out_path = os.getenv('WORKSPACE_BUCKET') + '/spark-test/outputs'
```
Comment (ericsong) [inline api/docs/pyspark-usage.md:133]: nit: `out_path` -> `gcs_out_path`

File: api/docs/pyspark-usage.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+count = 5
+
+def write(i):
+    path = f"my_output-{i}-of-{count}.txt"
```
Comment (ericsong) [inline api/docs/pyspark-usage.md:137]: nit: `path` -> `filename`

**Review by ericsong — COMMENTED** [review]
content looks good. I had a bunch of comments but they're all just in the context of making the sample code almost overly verbose and descriptive so our users have more confidence in knowing what's going on and how they could modify it.

File: api/docs/pyspark-usage.md (hunk 2 above)
Comment (calbach) [reply api/docs/pyspark-usage.md:91]: This is fairly common in bash scripting, and is way more ergonomic for writing a file from a shell cmd than echo. I suspect most users will either:
1. know the syntax, or don't know it, but are able to pick it up and extend it, or
2. will be confused by either this approach or the echo approach

I'm not sure there will be many inbetween. For the second camp, I added a comment about creating a script file via Jupyter.

File: api/docs/pyspark-usage.md (hunk 5 above)
Comment (calbach) [reply api/docs/pyspark-usage.md]: went with "bucket", rather than "gcs"

**Review by calbach — COMMENTED** [review]
thanks

**Review by ericsong — APPROVED** [review]

**Review by yonghaoy — APPROVED** [review]

**Pull request merged**